src/AggregateFunctions/Helpers.h

```diff
@@ -87,8 +87,8 @@ static IAggregateFunction * createWithNumericType(const IDataType & argument_typ
     return nullptr;
 }
 
-template <template <typename, typename> class AggregateFunctionTemplate, template <typename> class Data, typename... TArgs>
-static IAggregateFunction * createWithUnsignedIntegerType(const IDataType & argument_type, TArgs && ... args)
+template <template <typename, typename> class AggregateFunctionTemplate, template <typename> class Data, bool support_signed = false, typename... TArgs>
+static IAggregateFunction * createWithOptionSignedIntegerType(const IDataType & argument_type, TArgs && ... args)
```

Comment:
To retain an uniform interface across all helper functions in this header, please don't change `createWithUnsignedIntegerType()`. Instead, please add a function `createWithSignedIntegerType()` that wraps l. 103-109. And then please make a function `createWithIntegerType()` which calls the previous two functions using a check similar to l. 134.